onnxruntime/core/providers/cpu/reduction/reduction_ops.h

```diff
@@ -8,12 +8,33 @@
 #include "core/common/optional.h"
 #include "core/framework/op_kernel.h"
 #include "core/providers/cpu/containers.h"
+#include "core/util/math.h"
 #include "core/util/math_cpuonly.h"
 #include "core/platform/threadpool.h"
+#include "core/common/safeint.h"
 #include <cmath>
 
 namespace onnxruntime {
 
+typedef enum _FastReduceKindValues : uint8_t {
+  NONE = 0,   // no fast implementation
+  K = 1,      // kept dim = no reduce
+  R = 2,      // reduced dim = all reduced
+  KR = 4,     // kept dim, reduced dim
+  RK = 8,     // reduced dim, kept dim
+  KRK = 16,   // kept dim, reduced dim, kept dim
+  EMPTY = 32  // empty reduce
+} FastReduceKindValues;
+
+typedef uint8_t FastReduceKind;
+
```

Comment:
nit: as this is C++ these two 'typedef's could be combined into `enum class FastReduceKind`.

[https://google.github.io/styleguide/cppguide.html#Enumerator_Names](https://google.github.io/styleguide/cppguide.html#Enumerator_Names)

Reply:
done

Reply:
done